content/flux/v0.x/join-data/_index.md

```diff
@@ -0,0 +1,143 @@
+Learn how to use the `join` package to join two data sets with common values.
+
+The `join` package supports the following join methods:
+
+{{< flex >}}
+{{< flex-content "quarter" >}}
+<a href="#perform-an-inner-join">
+  <p style="text-align:center"><strong>Inner join</strong></p>
+  {{< svg svg="static/svgs/join-diagram.svg" class="inner small center" >}}
+</a>
+{{< /flex-content >}}
+{{< flex-content "quarter" >}}
+<a href="#perform-a-left-outer-join">
+  <p style="text-align:center"><strong>Left outer join</strong></p>
+  {{< svg svg="static/svgs/join-diagram.svg" class="left small center" >}}
+</a>
+{{< /flex-content >}}
+{{< flex-content "quarter" >}}
+<a href="#perform-a-right-outer-join">
+  <p style="text-align:center"><strong>Right outer join</strong></p>
+  {{< svg svg="static/svgs/join-diagram.svg" class="right small center" >}}
+</a>
+{{< /flex-content >}}
+{{< flex-content "quarter" >}}
+<a href="#perform-a-full-outer-join">
+  <p style="text-align:center"><strong>Full outer join</strong></p>
+  {{< svg svg="static/svgs/join-diagram.svg" class="full small center" >}}
+</a>
+{{< /flex-content >}}
+{{< /flex >}}
+
+- [How join functions work](#how-join-functions-work)
+  - [Input streams](#input-streams)
+  - [Join predicate function (on)](#join-predicate-function-on)
+  - [Join output function (as)](#join-output-function-as)
+- [Perform join operations](#perform-join-operations)
+  {{< children type="anchored-list" filterOut="Troubleshoot join operations" >}}
+- [Troubleshoot join operations](#troubleshoot-join-operations)
+
+## How join functions work
+
+All functions in the `join` package join _two_ streams of tables together based 
+on common values in each input stream.
+
+- [Input streams](#input-streams)
+- [Join predicate function (on)](#join-predicate-function-on)
+- [Join output function (as)](#join-output-function-as)
+
+### Input streams
+
+Each input stream is assigned to the `left` or `right` parameter.
+Input streams can be defined from any valid data source.
+For more information, see:
+
+- [Query data sources](/flux/v0.x/query-data/)
+- Define ad hoc tables with [`array.from()`](/flux/v0.x/stdlib/array/from/)
+
+#### Data requirements
+
+To join two streams of data with the `join` package, each stream must have:
+
+- **One or more columns with common values to join on**.  
+  Columns do not need identical labels, but they do need to have comparable values.
+- **Identical [group keys](/flux/v0.x/get-started/data-model/#group-key)**.  
+  Functions in the `join` package use group keys to quickly determine what tables
+  from each input stream should be paired and evaluated for the join operation.  
+  _Both input streams must have the same group key._
+  This likely requires using [`group()`](/flux/v0.x/stdlib/universe/group/)
+  to regroup each input stream before joining them together.
+
+  {{% note %}}
+Only tables with the same [group key instance](/flux/v0.x/get-started/data-model/#example-group-key-instances)
```

Comment:
I think there's some confusion over terms here. As I understand it, a group key is a specific set of values for a given set of columns. The group key columns are the columns we use to get the group key for a specific row or set of rows.

Maybe I'm misunderstanding something here. Just wanted to clarify.

Reply:
@scbrickley I've been trying to land on these semantics for a while and, after talking with Nathaniel, settled on this:

The **group key** is the columns by which a _stream_ is grouped  (e.g. `[_measurement, tag, _field`). A **group key instance** is the columns and their values associated with a specific group (table) in the stream (e.g. `[_measurement="m", tag="foo", _field="usage_user"]`).

Reply:
That seems like a fair definition. It's different from my understanding, but I suppose as long as the docs are internally consistent in how they use the terms that should be fine.